### PR TITLE
Add monitring overview to docs

### LINKED ISSUE #4337: More metrics about deployment workflow that be exported by Piped
**What would you like to be added**:

**Why is this needed**:

Though we already support some metrics export via piped admin server (at /metrics), users need more deployment metrics for a better monitoring board to watch piped operations.

### PR DESCRIPTION
**Pull request by kentakozuka**
**What this PR does / why we need it**:

Add docs to the metrics section

**Which issue(s) this PR fixes**:

Fixes #4337

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by kentakozuka** [comment]
@khanhtc1202 I just added docs to `docs-dev`. Should I add it to `docs` too?

**Comment by khanhtc1202** [comment]
> I just added docs to `docs-dev`. Should I add it to `docs` too?

Yes, please 👍 Also add that to the `/docs-v0.43.x` as well.

File: docs/content/en/docs-dev/user-guide/managing-controlplane/metrics.md [hunk 1]
```diff
@@ -9,6 +9,17 @@ description: >
...
 This page walks you through how to set up and use them.
 
+## Monitoring overview
+
+![](/images/metrics-architecture.png)
+<p style="text-align: center;">
+Monitoring Architecture
+</p>
+
+The piped agent collects its own monitoring information and periodically sends metrics to the Control plane, which publishes all of the resource usage and cluster information it collects and the metrics sent by the piped agent on its http server. When the PipeCD monitoring feature is turned on, Prometheus, Alertmanager and Grafana are deployed with the Control plane, and Prometheus retrieves metrics information from the Control plane's http server.
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-controlplane/metrics.md:19]:
```suggestion
The piped agent collects its metrics and periodically sends them to the Control plane. The Control plane then compacts its resource usage and cluster information with the metrics sent by the piped agent and re-publishes them via its admin server. When the PipeCD monitoring feature is turned on, Prometheus, Alertmanager, and Grafana are deployed with the Control plane, and Prometheus retrieves metrics information from the Control plane's admin server.
```

File: docs/content/en/docs-dev/user-guide/managing-controlplane/metrics.md [hunk 2]
```diff
@@ -9,6 +9,17 @@ description: >
...
+
+The piped agent not only sends the collected monitoring metrics to the Control plane, but also provides them through its internal http server, so that the developer managing the piped agent can also get metrics directly from the piped agent and monitor them with their custom monitoring service.
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-controlplane/metrics.md:21]:
```suggestion
The piped agent not only sends the collected monitoring metrics to the Control plane, but also provides them through its admin server. The developers managing the piped agent can also use exposed metrics and monitor them with their custom monitoring service.
```

**Comment by khanhtc1202** [comment]
Also, I don't remember correctly, but do we have any places which mention the `admin server` of pipecd components? If not, I guess we should add it so that users can understand how to get the metrics from pipecd's components.

**Comment by kentakozuka** [comment]
I couldn't find any places to mention "admin server" in documents.
What about just HTTP server? If we use "admin server", we need to explain a bit more about the role of "admin server".

**Comment by khanhtc1202** [comment]
> What about just HTTP server? If we use "admin server", we need to explain a bit more about the role of "admin server".

I think we should make it clear what is admin server so that piped managing dev can understand where they can actually get the piped metrics, mentioning as "HTTP server" could lead to a confusing situation.

**Comment by kentakozuka** [comment]
I got it 👍 I'll add the explanation about the "admin server".

**Comment by kentakozuka** [comment]
@khanhtc1202 Fixed 👍

File: docs/content/en/docs-dev/user-guide/managing-controlplane/metrics.md [hunk 3]
```diff
@@ -9,6 +9,19 @@ description: >
...
 This page walks you through how to set up and use them.
 
+## Monitoring overview
+
+![](/images/metrics-architecture.png)
+<p style="text-align: center;">
+Monitoring Architecture
+</p>
+
+Both the Control plane and piped agent have their own "admin servers", which are simple HTTP servers providing operational information such as health status, running version, go profile, and monitoring metrics.
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-controlplane/metrics.md:19]:
```suggestion
Both the Control plane and piped agent have their own "admin servers" (the default port number is 9085), which are simple HTTP servers providing operational information such as health status, running version, go profile, and monitoring metrics.
```

File: docs/content/en/docs-v0.43.x/user-guide/managing-controlplane/metrics.md [hunk 4]
```diff
@@ -9,6 +9,19 @@ description: >
...
 This page walks you through how to set up and use them.
 
+## Monitoring overview
+
+![](/images/metrics-architecture.png)
+<p style="text-align: center;">
+Monitoring Architecture
+</p>
+
+Both the Control plane and piped agent have their own "admin servers", which are simple HTTP servers providing operational information such as health status, running version, go profile, and monitoring metrics.
```
Comment (khanhtc1202) [inline docs/content/en/docs-v0.43.x/user-guide/managing-controlplane/metrics.md:19]:
```suggestion
Both the Control plane and piped agent have their own "admin servers" (the default port number is 9085), which are simple HTTP servers providing operational information such as health status, running version, go profile, and monitoring metrics.
```

File: docs/content/en/docs/user-guide/managing-controlplane/metrics.md [hunk 5]
```diff
@@ -9,6 +9,19 @@ description: >
...
 This page walks you through how to set up and use them.
 
+## Monitoring overview
+
+![](/images/metrics-architecture.png)
+<p style="text-align: center;">
+Monitoring Architecture
+</p>
+
+Both the Control plane and piped agent have their own "admin servers", which are simple HTTP servers providing operational information such as health status, running version, go profile, and monitoring metrics.
```
Comment (khanhtc1202) [inline docs/content/en/docs/user-guide/managing-controlplane/metrics.md:19]:
```suggestion
Both the Control plane and piped agent have their own "admin servers" (the default port number is 9085), which are simple HTTP servers providing operational information such as health status, running version, go profile, and monitoring metrics.
```

**Review by khanhtc1202 — COMMENTED** [review]
LGTM with small nits

**Comment by kentakozuka** [comment]
@khanhtc1202 Thanks! Can you approve it?

**Review by khanhtc1202 — APPROVED** [review]
👍

**Comment by khanhtc1202** [comment]
Approved, feel free to merge it on your pace.

**Pull request merged**